### ISSUE TITLE
Sourcemaps are broken in 0.8.0 (works fine in 0.7.2) when using with Vue SFC options api.

### ISSUE THREAD

**Issue by leo-buneev** [antfu/unplugin-auto-import#194]
Repro: https://stackblitz.com/edit/vite-zrayd3?file=src/App.vue
Click the button and look at reported line in browser console.

**Comment by asmodai1985** [comment]
I had the same trouble.It is resolved in 0.8.7

**Comment by luqiudi** [comment]
i had the same trouble in tsx files.
it broken sourcemap

**Comment by leo-buneev** [comment]
@antfu Actually found the issue! Seems to be simple.

on 0.7.2 it worked, because there was "sourcemap:" option in config that defaulted to false. And actually, you don't really need sourcemaps with this plugin, because it puts all injected imports into single line.

on 0.8.0 onwards, there is no "sourcemap" option, and plugin always generates sourcemaps with `magic-string` library. However, generated sourcemaps in my case are totally wrong, and to fix that it was enough to pass `hires: true` option to `s.generateMap()`. This way sourcemap is generated correctly.

BTW, exactly same problem was found in unplugin-vue-components.
Should I create a MR to add `hires: true`, or to add `sourcemap: true|false` back to config @antfu ?

**Comment by zam157** [comment]
@leo-buneev I had the same problem. Maybe antfu didn't notice this issue, you should PR directly

**Comment by superchangme** [comment]
I had the same problem,even i upgrade vite to 4.0 ,unplugin-auto-import unplugin-vue-components to newleast version debugger code position is still wrong:(

**Comment by whaaaley** [comment]
Yup, broken in both auto-import and components. How do people live without sourcemaps?

**Comment by whaaaley** [comment]
@leo-buneev Did you ever manage to resolve this issue? I noticed some things were merged since but the issue still persists for me.

**Comment by leo-buneev** [comment]
I've fixed it for my project in a fork. It's not kept up-to-date, though.

https://github.com/leo-buneev/unplugin-auto-import.git
https://github.com/leo-buneev/unplugin-vue-components.git
